image_client/casbotany_sql_lite.py

```diff
@@ -383,51 +385,83 @@ def table_sql_list():
 
     return sql_bot_list
 
+
 def casbotany_lite_creator():
-    connect = sqlite3.connect('cas_botanylite.db')
+    """casbotqny_lite_creator: casbotany_lite_creator: creates the
+                  sqllite tables contained in the sqllite DDL list"""
+    connect = sqlite3.connect('tests/casbotany_lite.db')
     sql_list = table_sql_list()
     curs = connect.cursor()
     # running a loop through tables for sql_lite
     for table in sql_list:
         curs.execute(table)
         connect.commit()
+
     # closing connection
     curs.close()
     connect.close()
 
 
 def sql_lite_connection(db_name):
+    """sql_lite_connection: creates the connection for sqllite db,
+        used as the "connection" parameter in other functions
+        args:
+            db_name: db_name is the file path to the sqlite db."""
     connection = sqlite3.connect(db_name)
     return connection
 
-def sql_lite_insert(sql, db_name):
-    """facimile statement to insert_table_record in sql_csv_utils.py"""
-    connect = sqlite3.connect(db_name)
-    curs = connect.cursor()
+
+def sql_lite_insert(sql, connection, logger_int):
+    """sql_lite_insert: facsimile to insert_table_record in sql_csv_utils.py
+        args:
+            sql: sql string to send to database
+            connection: the sqlite connection used to insert data
+            logger_int: the instance of logger to use for error reporting
+    """
+    curs = connection.cursor()
     try:
         curs.execute(sql)
     except Exception as e:
-        raise ValueError(f"Exception thrown while processing sql: {sql}\n{e}\n")
+        logger_int.error(f"Exception thrown while processing sql: {sql}\n{e}\n")
     try:
-        connect.commit()
+        connection.commit()
 
     except Exception as e:
         raise ValueError(f"sql debug: {e}")
 
     curs.close()
-    connect.close()
+    connection.close()
 
-def casbotany_lite_getrecord(sql_query: str):
-    """modified get one record function for sql lite"""
-    connect = sqlite3.connect('cas_botanylite.db')
-    curs = connect.cursor()
+
+def casbotany_lite_getrecord(id_col, tab_name, key_col, match, match_type="string"):
```

Comment:
added comments, and  a sql parsing section in order to have similar arguments to get_one_match in sql_csv_utils, so I could easily monkeypatch between the two.